### PR TITLE
Add Initial Parser for Window Sysmon Event 3

### PR DESCRIPTION
**Pull request by goosvorbook**
Required items, please complete
   
   Change(s):
    - Added ASIM parser for Sysmon Event 3 ( both Event and WindowsEvent)

   Reason for Change(s):
   - New addition.

   Version Updated:
   - Non-existent to 0.0.1 

   Testing Completed:
   - See guidance below

### CODE REVIEW

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        DstProcessGuid = ProcessGuid,
+        DstProcessName = Image
+| project-away outbound, ProcessId, ProcessGuid, Image
+;
+union InboundNetworkEvents,OutboundNetworkEvents
+        | project-rename 
+            EventStartTime = UtcTime,
+            Dvc = Computer,
+            Hostname = DestinationHostname,
+            IpAddr = SourceIp,
```
Comment (oshezaf) [inline Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:106]: While correct, I feel it is somewhat confusing. I would project-rename to DstIpAddr and SrcIpAddr and have the others as aliases using extend later on.

**Review by oshezaf — CHANGES_REQUESTED** [review]

**Comment by goosvorbook** [comment]
@microsoft-github-policy-service agree [company="{Wortell}"]


Met vriendelijke groet / Kind regards,​​
Guus Verbeek
.
Bekijk
​ook eens:
***@***.***<https://www.wortell.nl/nl/themas/secure>
Consultant
.
+31611084192<tel:+31611084192>

**Comment by goosvorbook** [comment]
@microsoft-github-policy-service agree company="Wortell"


Met vriendelijke groet / Kind regards,​​
Guus Verbeek
.
Bekijk
​ook eens:
***@***.***<https://www.wortell.nl/nl/themas/secure>
Consultant
.
+31611084192<tel:+31611084192>

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            SourcePortName = tostring(EventData.SourcePortName),
+            DestinationIsIpv6 = tobool(EventData.DestinationIsIpv6),
+            DestinationIp = tostring(EventData.DestinationIp),
+            DestinationHostname = tostring(EventData.DestinationHostname),
+            DestinationPort = toint(EventData.DestinationPort),
+            DestinationPortName = tostring(EventData.DestinationPortName)
+    | extend outbound = SourcePort > DestinationPort
+    | project-away EventData
+};
+let OutboundNetworkEvents =  union Sysmon3_Event, Sysmon3_WindowsEvent
```
[the remark this answers is not in the document]
Comment (goosvorbook) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:81]: I can probably do something like SourceIp in ('127.0.0.1', '0.0.0.0')
But these datafeeds do not contain HostIP

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            EventSeverity = 'Informational',
+            DvcOs = 'Windows',
+            Protocol = toupper(Protocol),
+            EventOriginalType = '3' // Set with a constant value to avoid parsing
+        | extend // aliases
+            DvcDomain = iff(tostring(split(User, "\\")[0]) == Dvc, " ", tostring(split(User, "\\")[0])),
+            SrcDomain = iff(tostring(split(User, "\\")[0]) == Dvc, " ", tostring(split(User, "\\")[0])),
+            DvcDomainType = "Windows",
+            SrcDomainType = "Windows",
+            ASimMatchingHostname = "Both",
```
[the remark this answers is not in the document]
Comment (goosvorbook) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:135]: Can you specify which two are not needed of these 4

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (oshezaf) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:81]: Indeed. It seems that Sysmon behaved differently on Linux and Windows. Actually, there is no easy way, as the source port is already defined as source port. Comparing to destination port does not provide more information. Our challenge in this case is limited to correctly assigning the process and user fields. 

Two heuristics that work for some cases are:
- Check which address is private. If only one address is private, it is the local machine, and they user and process fields should assume its role. 
- If either the source or destination hostname are empty ("-"), the other one is the local machine.

If both fail, let's just put the user and process fields on source. It is not correct, but I can't think of something else.

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (oshezaf) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml:134]: The comment is always on the last line. GitHub just displays 4 lines for your convenience. Always, ASimMatchingHostname, And ASimMatchingIpAddr

**Pull request closed without merging**